sql/mysqld.cc

```diff
@@ -7957,6 +7960,34 @@ static int show_connection_errors_net_ER_NET_WRITE_INTERRUPTED(THD *, SHOW_VAR *
   return 0;
 }
 
+static inline ulonglong get_super_read_only_block_microsec(
+    std::atomic_ullong &lock_block_timer) {
+  ulonglong lock_wait_microsec =
+      lock_block_timer.load(std::memory_order_seq_cst);
```

Comment:
This could be memory_order_relaxed, couldn't it?

Reply:
@laurynas-biveinis We could only if we don't care too much about accuracy of these counters. In other words, with `memory_order_relaxed` here, there is a chance that even when `init_global_rolock_timer` / `init_commit_lock_timer` are set inside one thread in `MDL_context::acquire_lock()` with `memory_order_seq_cst` (which is a default memory order for `std::atomic<>::operator =` ), another thread executing `show status like 'super_read_only_block_microsec'` may read these values as zeros.
I am not 100% sure that this is unacceptable, but personally, I would go with Sequentially-consistent memory order here for simplicity.
In theory, the most correct way to do this would be using `std::memory_order_release` for setting these vars inside `MDL_context::acquire_lock()` and reading them here with `std::memory_order_acquire`, but because the order in which they are set is not determined (`init_global_rolock_timer` is set in one call of `MDL_context::acquire_lock()` whereas `init_commit_lock_timer` in another), I wouldn't go this road.

So, just to simplify
```suggestion
      lock_block_timer;
```

Reply:
BTW, I'have recently discovered a very nice tool "CppMem: Interactive C/C++ memory model"
http://svr-pes20-cppmem.cl.cam.ac.uk/cppmem/
which is supposed to statically check race conditions for cases like this.
However, it's very sophisticated and I must admit I understand only 10% of it :)